src/transaction/requests.rs

```diff
@@ -445,11 +445,13 @@ impl Merge<ResponseWithShard<kvrpcpb::PessimisticLockResponse, Vec<kvrpcpb::Muta
 
 pub fn new_scan_lock_request(
     start_key: Vec<u8>,
+    end_key: Vec<u8>,
     safepoint: u64,
     limit: u32,
 ) -> kvrpcpb::ScanLockRequest {
     let mut req = kvrpcpb::ScanLockRequest::default();
     req.set_start_key(start_key);
+    req.set_end_key(end_key);
```

Comment:
The `impl Shardable`, `impl HasNextBatch`, and `impl NextBatch` below also need some changes.

Reply:
Range is added in `impl Shardable`. `TODO` is added in `impl HasNextBatch`, as it's not easy to get the request range in `ScanLockResponse`. It does not scan useless regions as `has_more_batch` will be judged before another scan. PTAL~